File: string/validPalindrome.py
# https://leetcode.com/problems/valid-palindrome/
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        a = re.sub(r"\W", r"", s.lower())
        return a == a[::-1]

# Without regexp
# def isPalindrome(self, s):
#         s = [c.lower() for c in s if c.isalnum()]
#         return s == s[::-1]

File: string/validPalindrome.js
// https://leetcode.com/problems/valid-palindrome/
/**
 * @param {string} s
 * @return {boolean}
 */
const isPalindrome = (s) => {
  const strippedString = s.replace(/\W/g, "");
  const reversedString = strippedString.split("").reverse().join("");

  return strippedString.toLowerCase() == reversedString.toLowerCase();
};
